cmd/blockhash/main.go

```diff
@@ -235,6 +235,10 @@ func (b *hashBuilder) ftype(structName, s string, expr ast.Expr, directives map[
 		return "uint64(" + s + ")", 8
 	case "Block":
 		return "world.BlockHash(" + s + ")", 32
+	case "ButtonType":
+		return "uint64(" + s + ".Uint8())", 6
+	case "PressurePlateType":
+		return "uint64(" + s + ".Uint8())", 8
```

Comment:
If I'm not mistaken, you can reduce this to 2 for buttons and 3 for pressure plates